R/covr.R

```diff
@@ -505,17 +505,25 @@ run_commands <- function(pkg, lib, commands) {
 # regardless of how the process terminates.
 # @param pkg_name name of the package to add hooks to
 # @param lib the library path to look in
+# @param trace_dir the directory where trace files will be saved; this can be
+#        an \code{expression} expression to evaluate the call at package load
+#        time, e.g. \code{expression(Sys.getenv("COVERAGE_DIR"))}.
 # @param fix_mcexit whether to add the fix for mcparallel:::mcexit
-add_hooks <- function(pkg_name, lib, fix_mcexit = FALSE) {
+add_hooks <- function(pkg_name, lib, trace_dir, fix_mcexit = FALSE) {
```

Comment:
I would be ok with setting the default of trace_dir to be `trace_dir = Sys.getenv("COVERAGE_DIR", lib)`. Which would give you the behavior you were looking for just by setting the `COVERAGE_DIR` environment variable.

Reply:
Done. Thank you.